### ISSUE TITLE
Is it possible to quickly assess is tree has changed ?

### ISSUE THREAD

**Issue by vladp** [joaonuno/tree-model-js#57]
Hello,
my code is using your library extensively. Thank you for making it available.

I use this component within React application.
For some UI things I do walk the tree walk on every re-render event.
When the number of nodes, typically does not exceeds 30 or 40, it works reasonably fast.

However, I am anticipating (this is somewhat optimistic) that the number of nodes will be a couple of 100s.
combining it with the number of re-renders a non-trivial React UI will cause, I will be traversing this tree several times per second in some instances. And since some of these screens will be running on low-powered devices within ancient web-browsers , the cost becomes non-trivial.

Therefore, i would like to cache the walk (my walks produce basically nodes ordered in pre-order).

To do the caching, I am thinking either function memoisation or just simple 'rootnode -> preorder walk result array'  key-val cache.
To implement those, I would like to be able to quickly assess if the tree underneath my cache has changed or not.

In other words, if there were adds/deletes to the hierarchy itself (I do no care if the model attributes were updated or not).

Therefore, I was wondering if there is something internal (or can be exposed) that I could use.
May be some sort of a bitmap or a hash at the root that is updated every time hierarchy changes?

**Comment by joaonuno** [comment]
There is nothing built-in to give you that. My suggestion is to keep that info in your code and update it every time a tree altering method is called.

**Comment by vladp** [comment]
ok. thank you. will do that.
